doc/nat/nat_design_spec.md

```diff
@@ -0,0 +1,1422 @@
+If a matching SNAT entry or DNAT entry does not exist in the hardware, the packets are trapped to CPU as NAT miss packets to be processed by the kernel software forwarding path.
+
+Corresponding to the ACL, the equivalent deny and permit rules are added in the iptables so that the intended NAT miss packets are dynamically SNAT'ted in the kernel.
+
+When the ACL that is bound to the NAT pool is deleted, the corresponding iptables SNAT rules are deleted in the kernel.
+When the ACL rules are created or modified, the corresponding iptables SNAT rules are updated in the kernel.
+Multiple ACLs can be bound to a single NAT pool. 1 ACL can be bound atmost to a single NAT pool.
+
+If there is no matching ACL configured, it is treated as 'implicit permit all' and hence the corresponding NAT pool in the binding is applied on all the traffic.
+
+Following match actions of the ACL are handled to match against the traffic for dynamic SNAT mapping:
+- Source IP address or Subnet
+- Destination IP address or Subnet
+- Source L4 port or L4 port range
+- Destination L4 port or L4 port range
+- IP protocol
+
+#### 3.3.1.2 IP Interface config events
+When the IP on the outside NAT zone interface is deleted, for any matching NAT pool and the matching Static NAT/NAPT entries, the corresponding iptables SNAT rules and the Static DNAT rules are deleted in the kernel.
+The conntrack entries in the kernel and the APP_DB entries are aged out eventually by the inactivity period timeouts since there will not be any NAT traffic crossing over the deleted IP interfaces.
+
+#### 3.3.1.3 Interface link events
+When the outside NAT zone interface's link goes down, the iptables NAT rules, the conntrack entries and the APP_DB dynamic entries are not removed immediately.
+
+This is done for the reasons below:
+
+	- The temporary link flap events happening on the NAT outside interface should not
+	  affect the NAT translation entries (corresponding to the end to end TCP/UDP sessions).
+	- To avoid performance issues during link flaps.
+	- The dynamic entries are aged out eventually due to inactivity timeout in case
+	  the NAT outside interface is link down for more than the timeout period.
+
+### 3.3.2 Natsync daemon
+
+NatSyncd listens to the conntrack netlink notification events from the kernel for the creation, update and removal events of the connections in the conntrack table.
+Once the notifications for SNAT and DNAT entries are received, they are pushed into the NAT_TABLE/NAPT_TABLE/NAPT_TWICE_TABLE in the APP_DB to be consumed by the NatOrch Agent. More details in the section 3.4.
+
+### 3.3.3 NatOrch Agent
+NAT feature is disabled by default.
+When the NAT feature is enabled, NatOrch enables the feature and the NAT miss trap actions in the hardware.
+The SNAT or DNAT miss packets are rate limited to CPU @ 600pps.
```

Comment:
How to support rate limit? let's say user wanted to rate limit 1K flows.

Reply:
Through Copp configuration. Please refer to earlier response above.